src/sqlancer/feldera/ast/FelderaConstant.java

```diff
@@ -18,40 +22,100 @@ private static double round(double number, int places) {
         return decimal.doubleValue();
     }
 
-    public static FelderaConstant getRandomConstant(FelderaGlobalState globalState,
-            FelderaSchema.FelderaDataType type) {
-        switch (type) {
+    public static FelderaExpression getRandomConstant(FelderaGlobalState globalState,
+            FelderaSchema.FelderaCompositeDataType type) {
+        switch (type.getPrimitiveType()) {
         case BOOLEAN:
             return new FelderaBooleanConstant(Randomly.getBoolean());
-        case TINYINT:
-            return FelderaIntConstant.getRandom(globalState, 8);
-        case SMALLINT:
-            return FelderaIntConstant.getRandom(globalState, 16);
         case INT:
-            return FelderaIntConstant.getRandom(globalState, 32);
-        case BIGINT:
-            return FelderaIntConstant.getRandom(globalState, 64);
+            return FelderaIntConstant.getRandom(globalState, type.getSize());
         case VARCHAR:
-            return FelderaVarcharConstant.getRandom(globalState);
+            return FelderaVarcharConstant.getRandom(globalState, type.getSize());
         case CHAR:
-            return FelderaCharConstant.getRandom(globalState);
+            return FelderaCharConstant.getRandom(globalState, type.getSize());
         case NULL:
             return new FelderaNullConstant();
         case TIME:
-            return FelderaTimeConstant.getRandom(globalState);
+            return new FelderaCast(FelderaTimeConstant.getRandom(globalState), type);
```

Comment:
is this because now time literals are strings?

Reply:
Yes.